mllib/src/main/scala/org/apache/spark/ml/classification/MultilayerPerceptronClassifier.scala

```diff
@@ -81,7 +83,8 @@ private[classification] trait MultilayerPerceptronParams extends PredictorParams
   final val solver: Param[String] = new Param[String](this, "solver",
     "The solver algorithm for optimization. Supported options: " +
       s"${MultilayerPerceptronClassifier.supportedSolvers.mkString(", ")}. (Default l-bfgs)",
-    ParamValidators.inArray[String](MultilayerPerceptronClassifier.supportedSolvers))
+    (value: String) => MultilayerPerceptronClassifier.supportedSolvers
+      .contains(value.toLowerCase(Locale.ROOT)))
```

Comment:
What do you think of adding a new function in ```object ParamValidators``` as
```
def inStringArray(allowed: Array[String]): String => Boolean = { (value: String) =>
    allowed.contains(value.toLowerCase(java.util.Locale.ROOT))
  }
```
to facilitate similar check here and other place.